internal/server/pagination.go

```diff
@@ -31,6 +33,6 @@ func PaginationToResponse(p data.Pagination) api.PaginationResponse {
 		Page:       p.Page,
 		Limit:      p.Limit,
 		TotalCount: p.TotalCount,
-		TotalPages: p.TotalPages,
+		TotalPages: int(math.Ceil(float64(p.TotalCount) / float64(p.Limit))),
```

Comment:
The `TotalPages` field is only used in the API response. We don't really have any use for it internally. So I removed the field from the internal struct, and moved the calculation here. I think the api handler code is the right place for it, because the value is only used in the API.